conformance/conformance.go

```diff
@@ -214,9 +215,15 @@ func RunConformanceWithOptions(t *testing.T, opts confsuite.ConformanceOptions)
 	cSuite.Setup(t, tests.ConformanceTests)
 
 	sharedGwNN := types.NamespacedName{Name: SharedGatewayName, Namespace: SharedGatewayNamespace}
+	secondaryGwNN := types.NamespacedName{Name: SecondaryGatewayName, Namespace: SharedGatewayNamespace}
 
-	// Validate Gateway setup.
+	// Validate Gateway setup for both Gateways.
+	t.Logf("Validating primary Gateway setup: %s/%s", sharedGwNN.Namespace, sharedGwNN.Name)
 	ensureGatewayAvailableAndReady(t, cSuite.Client, opts, sharedGwNN)
```

Comment:
nit: how long does ensureGatewayAvailableAndReady() typically take?  Wondering if we should just do them in parallel.

Something like this:

```
var wg wync.WaitGroup

for _, gwNN := range []types.NamespacedName{sharedGwNN, secondaryGwNN} {
    wg.Add(1)
    go func() {
        defer wg.Done()
        ensureGatewayAvailableAndReady(t, cSuite.Client, opts, gwNN)
    }()
}
wg.Wait()
```

Reply:
Good point, both gateways are introduced in the `conformance/resources/manifests/manifests.yaml` and via the test suite setup here 'https://github.com/kubernetes-sigs/gateway-api/blob/b7d2c5788bf38fc2c18085de524e204034c69a14/conformance/utils/suite/suite.go#L355

which underneath uses https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/utils/kubernetes/apply.go#L250 This method does apply (create/update) the changes sequentially. 

This could be an optimization to consider for that method. In the ensureGatewayAvailableAndReady we only pull the status to see if the gateway is ready, so checking the status sequentially does n't add a significant overhead.

Reply:
Ah okay got it, thanks!